src/Employer/Employer.Web/Orchestrators/LegalEntityAgreementOrchestrator.cs

```diff
@@ -26,18 +26,19 @@ public class LegalEntityAgreementOrchestrator
         public async Task<LegalEntityAgreementSoftStopViewModel> GetLegalEntityAgreementSoftStopViewModelAsync(
             VacancyRouteModel vrm, long? selectedLegalEntityId)
         {
-            var vacancy = 
-                await Utility.GetAuthorisedVacancyForEditAsync(
+            var vacancy = await
+                Utility.GetAuthorisedVacancyForEditAsync(
                     _client, _vacancyClient, vrm, RouteNames.LegalEntityAgreement_SoftStop_Get);
 
             var legalEntityId = selectedLegalEntityId.HasValue ? selectedLegalEntityId.Value : vacancy.LegalEntityId;
+            var legalEntity = await _legalEntityAgreementService.GetLegalEntityAsync(vrm.EmployerAccountId, legalEntityId);
             
             return new LegalEntityAgreementSoftStopViewModel
             {                
                 HasLegalEntityAgreement = 
                     await _legalEntityAgreementService.HasLegalEntityAgreementAsync(
                         vacancy.EmployerAccountId, legalEntityId),
-                LegalEntityName = vacancy.EmployerName,
+                LegalEntityName = legalEntity.Name,
```

Comment:
Why do we need a lookup? Can we use vacancy.LegalEntityName?

Reply:
oh..sweet!